### PR TITLE
Sorting My videos

### PR DESCRIPTION
**Pull request by eldiegod**
Issue: #492

"rebased" sorting PR

### CODE REVIEW

File: src/api/client/cache.ts [hunk 1]
```diff
@@ -84,16 +86,34 @@ const queryCacheFields: CachePolicyFields<keyof Query> = {
...
+            .sort(
+              (a, b) =>
+                (readField('createdAt', b.node) as Date).getTime() - (readField('createdAt', a.node) as Date).getTime()
+            )
+        : filteredEdges
+            ?.slice()
+            .sort(
+              (a, b) =>
+                (readField('createdAt', a.node) as Date).getTime() - (readField('createdAt', b.node) as Date).getTime()
+            )
```
Comment (kdembler) [inline src/api/client/cache.ts]: I think we could simplify that, we basically repeat the same snippet. Possibly we could just do regular sorting once and `reverse` if needed?

File: src/api/client/cache.ts [hunk 2]
```diff
@@ -84,16 +86,34 @@ const queryCacheFields: CachePolicyFields<keyof Query> = {
...
     ...relayStylePagination(getVideoKeyArgs),
-    read(existing: VideoConnection, opts) {
-      const isPublic = opts.args?.where?.isPublic_eq
-      const filteredEdges = existing?.edges.filter(
-        (edge) => opts.readField('isPublic', edge.node) === isPublic || isPublic === undefined
-      )
+    read(
+      existing: VideoConnection,
+
+      { args, readField }: { args: GetVideosQueryVariables | null; readField: ReadFieldFunction }
```
Comment (kdembler) [inline src/api/client/cache.ts]: I think typing should be `GetVideosConnectionQueryVariables` instead?

File: src/views/studio/MyVideosView/MyVideosView.tsx [hunk 3]
```diff
@@ -199,6 +219,20 @@ export const MyVideosView = () => {
           <>
             <TabsContainer>
               <Tabs initialIndex={0} tabs={mappedTabs} onSelectTab={handleSetCurrentTab} />
+              <SortContainer>
+                <Text variant="body2">Sort by</Text>
+                <Select
+                  helperText={null}
+                  value={sortVideosBy}
+                  items={[...SORT_OPTIONS]}
```
Comment (kdembler) [inline src/views/studio/MyVideosView/MyVideosView.tsx]: Let's remove that destructuring here. If needed let's remove `as const` as well, we can use the type of value directly (`VideoOrderByInput`)

File: src/views/studio/MyVideosView/MyVideosView.tsx [hunk 4]
```diff
@@ -199,6 +219,20 @@ export const MyVideosView = () => {
...
+                  onChange={(value) => {
+                    if (value) {
+                      setSortVideosBy(value)
+                      refetch({ orderBy: value })
+                    }
+                  }}
```
Comment (kdembler) [inline src/views/studio/MyVideosView/MyVideosView.tsx]: Let's move that to a named handler above `return`

File: src/api/client/cache.ts (hunk 2 above)
Comment (eldiegod) [reply src/api/client/cache.ts]: oh, yes thats right

**Review by kdembler — APPROVED** [review]

**Pull request merged**